### PR TITLE
[API] Mark tasks as mandatory bonus/additional etc

### LINKED ISSUE #705: [API] Mark tasks as "mandatory", "bonus/additional", etc.
The API should be extended to allow for the storage of a property which contains the information if the task is "mandatory", "bonus/additional", etc.

---
required for #704

### PR DESCRIPTION
**Pull request by pdrtzk**
closes #705

### CODE REVIEW

**Comment by pdrtzk** [comment]
Issue #704 can be implemented once this PR is merged.

File: modules/fbs-core/api/src/main/resources/migrations/06_task_requirement_type.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+BEGIN;
+
+alter table task
+    add requirement_type VARCHAR(200) not null;
```
Comment (Zitrone44) [inline modules/fbs-core/api/src/main/resources/migrations/06_task_requirement_type.sql:4]: * do we really need 200 characters here?
* a default value should be set to have already existing Task return the correct `requirementType` (`DEFAULT 'mandatory'`)

File: modules/fbs-core/api/src/main/scala/de/thm/ii/fbs/controller/TaskController.scala [hunk 2]
```diff
@@ -157,9 +157,10 @@ class TaskController {
         body.retrive("deadline").asText(),
         body.retrive("mediaType").asText(),
         body.retrive("description").asText(),
-        body.retrive("mediaInformation").asObject()
+        body.retrive("mediaInformation").asObject(),
+        body.retrive("requirementType").asText()
```
Comment (Zitrone44) [inline modules/fbs-core/api/src/main/scala/de/thm/ii/fbs/controller/TaskController.scala:161]: maybe we should validate here if `requirementType` has a valid value.

**Review by Zitrone44 — CHANGES_REQUESTED** [review]
Works :+1:

I would suggest that we validate `requirementType` to be either `madatory` or `optional`/`bonus`. And a few other suggestions:

**Review by Zitrone44 — APPROVED** [review]

**Pull request merged**